### PR TITLE
Service Registration: User Notification

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the user notification service and creating it.Please see #7839 for more context.

/to @choumx @jridgewell

### CODE REVIEW

File: extensions/amp-user-notification/0.1/amp-user-notification.js [hunk 1]
```diff
@@ -436,20 +436,19 @@ export class UserNotificationManager {
 /**
  * @param {!Window} window
  * @return {!UserNotificationManager}
- * @private
  */
-function getUserNotificationManager_(window) {
-  return fromClass(window, 'userNotificationManager',
-      UserNotificationManager);
+export function userNotificationManagerFor(window) {
+  installUserNotificationManager(window);
```
Comment (choumx) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: Remove this line and add `installUserNotificationManager` where needed.

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 2]
```diff
@@ -97,9 +98,11 @@ describe('amp-analytics', function() {
       windowApi = iframe.win;
       ampdoc = new AmpDocSingle(windowApi);
       cidServiceForDocForTesting(ampdoc);
-      uidService = installUserNotificationManager(iframe.win);
-
       ins = instrumentationServiceForDocForTesting(ampdoc);
+      installUserNotificationManager(iframe.win);
+      userNotificationManagerFor(iframe.win).then(manager => {
+        uidService = manager;
```
Comment (choumx) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: Is there any guarantee this will be set before `beforeEach` resolves?

Comment (kmh287) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: done

File: extensions/amp-user-notification/0.1/amp-user-notification.js [hunk 3]
```diff
@@ -107,7 +107,9 @@ export class AmpUserNotification extends AMP.BaseElement {
     this.urlReplacements_ = urlReplacementsForDoc(ampdoc);
     this.storagePromise_ = storageForDoc(ampdoc);
     if (!this.userNotificationManager_) {
-      this.userNotificationManager_ = getUserNotificationManager_(this.win);
+      installUserNotificationManager(window);
+      this.userNotificationManager_ = getService(window,
+                                                 'userNotificationManager');
```
Comment (choumx) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: Nit: +4 spaces from start of previous line. Sorry, annoying I know.

File: extensions/amp-user-notification/0.1/amp-user-notification.js [hunk 4]
```diff
@@ -435,21 +437,11 @@ export class UserNotificationManager {
...
-/**
- * @param {!Window} window
- * @return {!UserNotificationManager}
- * @private
  */
 export function installUserNotificationManager(window) {
-  return getUserNotificationManager_(window);
+  registerServiceBuilder(window,
+                         'userNotificationManager',
+                         UserNotificationManager);
```
Comment (choumx) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: Ditto.

File: extensions/amp-user-notification/0.1/amp-user-notification.js (hunk 3 above)
Comment (kmh287) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: done

File: extensions/amp-user-notification/0.1/amp-user-notification.js (hunk 4 above)
Comment (kmh287) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: done

**Pull request merged**